sklearn/model_selection/tests/test_search.py

```diff
@@ -1311,27 +1311,30 @@ def test_fit_grid_point():
     cv = StratifiedKFold()
     svc = LinearSVC(random_state=0)
     scorer = make_scorer(accuracy_score)
-
-    for params in ({'C': 0.1}, {'C': 0.01}, {'C': 0.001}):
-        for train, test in cv.split(X, y):
-            this_scores, this_params, n_test_samples = fit_grid_point(
-                X, y, clone(svc), params, train, test,
-                scorer, verbose=False)
-
-            est = clone(svc).set_params(**params)
-            est.fit(X[train], y[train])
-            expected_score = scorer(est, X[test], y[test])
-
-            # Test the return values of fit_grid_point
-            assert_almost_equal(this_scores, expected_score)
-            assert params == this_params
-            assert n_test_samples == test.size
-
-    # Should raise an error upon multimetric scorer
-    assert_raise_message(ValueError, "For evaluating multiple scores, use "
-                         "sklearn.model_selection.cross_validate instead.",
-                         fit_grid_point, X, y, svc, params, train, test,
-                         {'score': scorer}, verbose=True)
+    msg = (
+        "fit_grid_point is deprecated in version 0.23 "
+        "and will be removed in version 0.25")
+    with pytest.warns(FutureWarning, match=msg):
+        for params in ({'C': 0.1}, {'C': 0.01}, {'C': 0.001}):
+            for train, test in cv.split(X, y):
+                this_scores, this_params, n_test_samples = fit_grid_point(
```

Comment:
Move the `pytest.warns` here to be specific where the warning comes from.

```py
with pytest.warns(...):
	this_scores, this_params, =
```

Reply:
It will also raise deprecation warning on this code
```# Should raise an error upon multimetric scorer
        assert_raise_message(ValueError, "For evaluating multiple scores, use "
                             "sklearn.model_selection.cross_validate instead.",
                             fit_grid_point, X, y, svc, params, train, test,
                             {'score': scorer}, verbose=True)
```

should I applied `pytest.warns(FutureWarning, match=msg)` twice? (1 inside training loop and another before `assert_raise_message`)

Reply:
Lets not change this test and ignore the `FutureWarning`:


```py
# FIXME Remove when fit_grid_point is deprecated in 0.25
@ignore_warnings(category=FutureWarning)
def test_fit_grid_point():
    ...
```

And add another tests that only tests the deprecation:

```py
# FIXME Remove when fit_grid_point is deprecated in 0.25
def test_fit_grid_point_deprecated():
    with pytest.warns(FutureWarning, ...):
		...
```

Reply:
done